File: Java/app/src/main/java/com/github/shin1x1/jsonparser/lexer/Token.java
package com.github.shin1x1.jsonparser.lexer;

import java.math.BigDecimal;

public sealed interface Token {
    record Null() implements Token {
    }

    record True() implements Token {
    }

    record False() implements Token {
    }

    record LeftBrace() implements Token {
    }

    record RightBrace() implements Token {
    }

    record LeftBracket() implements Token {
    }

    record RightBracket() implements Token {
    }

    record Colon() implements Token {
    }

    record Comma() implements Token {
    }

    record Number(BigDecimal value) implements Token {
        public Number(java.lang.String value) {
            this(new BigDecimal(value));
        }
    }

    record String(java.lang.String value) implements Token {
    }
}

File: kotlin/src/main/kotlin/lexer/Token.kt
package lexer

sealed interface Token {
    object Null : Token {
        override fun toString(): kotlin.String = "Null"
    }

    object True : Token {
        override fun toString(): kotlin.String = "True"
    }

    object False : Token {
        override fun toString(): kotlin.String = "False"
    }

    object Colon : Token {
        override fun toString(): kotlin.String = "Colon"
    }

    object Comma : Token {
        override fun toString(): kotlin.String = "Comma"
    }

    object LeftBrace : Token {
        override fun toString(): kotlin.String = "LeftBrace"
    }

    object RightBrace : Token {
        override fun toString(): kotlin.String = "RightBrace"
    }

    object LeftBracket : Token {
        override fun toString(): kotlin.String = "LeftBracket"
    }

    object RightBracket : Token {
        override fun toString(): kotlin.String = "RightBracket"
    }

    data class Number(val value: Double) : Token {
        override fun toString(): kotlin.String = "Number:$value"
    }

    data class String(val value: kotlin.String) : Token {
        override fun toString(): kotlin.String = "String:$value"
    }
}
